src/wallets/encodedSeed.ts

```diff
@@ -0,0 +1,88 @@
+import { BigNumber } from '@waves/bignumber';
+import { SeedAdapter, TSignData } from '@waves/signature-adapter';
+import { customData, wavesAuth } from '@waves/waves-transactions';
+import * as libCrypto from '@waves/ts-lib-crypto';
+import * as create from 'parse-json-bignumber';
+import { Account, NetworkName } from 'accounts/types';
+import { Wallet } from './wallet';
+import { convertInvokeListWorkAround } from './utils';
+
+const { stringify } = create({ BigNumber });
+
+export interface EncodedSeedWalletInput {
+  encodedSeed: string;
+  name: string;
+  network: NetworkName;
+  networkCode: string;
+}
+
+interface EncodedSeedWalletData extends Account {
+  encodedSeed: string;
+}
+
+export class EncodedSeedWallet extends Wallet<EncodedSeedWalletData> {
+  private readonly _adapter: SeedAdapter;
+
+  constructor({
+    encodedSeed,
+    name,
+    network,
+    networkCode,
+  }: EncodedSeedWalletInput) {
+    const encodedSeedWithoutPrefix = encodedSeed.replace(/^base58:/, '');
+    const decodedSeed = libCrypto.base58Decode(encodedSeedWithoutPrefix);
+
+    super({
+      address: libCrypto.address(decodedSeed, networkCode),
+      encodedSeed: encodedSeedWithoutPrefix,
+      name,
+      network,
+      networkCode,
+      publicKey: libCrypto.publicKey(decodedSeed),
+      type: 'encodedSeed',
+    });
+
+    this._adapter = new SeedAdapter(
+      `base58:${encodedSeedWithoutPrefix}`,
+      networkCode
+    );
+  }
+
+  getSeed(): string {
+    throw new Error('Cannot get seed');
+  }
+
+  getEncodedSeed() {
+    return this.data.encodedSeed;
+  }
+
+  getPrivateKey() {
+    return libCrypto.privateKey(libCrypto.base58Decode(this.data.encodedSeed));
+  }
```

Comment:
Seems like it should be written like this
```js
getSeed(): string {
  return libCrypto.base58Decode(this.getEncodedSeed())
}

getEncodedSeed() {
  return this.data.encodedSeed;
}

getPrivateKey() {
  return libCrypto.privateKey(this.getSeed());
}
 ```

Reply:
`base58Decode` returns `Uint8Array` and encoded seed can contain arbitrary binary data. That's why `getSeed` throws for now. Not sure if we should check the contents and decide to throw or not. In UI we only allow to copy encoded seed back